Tests/Controller/ApiControllerTest.php

```diff
@@ -686,15 +689,256 @@ public function testProjectStats()
+                ],
+                [
+                    'date' => '2020-08-13',
+                    'durationSum' => 0,
+                ],
+            ],
+            $response['jobs']
+        );
+        $this->assertEquals(200, $frameworkClient->getResponse()->getStatusCode());
+
+        // fourth test - negative range
+        $frameworkClient->request(
+            'GET',
+            '/docker/stats/project/daily?fromDate=2022-08-09&toDate=2020-08-13&timezoneOffset=%2B00:00',
+            [],
+            [],
+            ['HTTP_X-StorageApi-Token' => STORAGE_API_TOKEN]
+        );
+        $response = json_decode($frameworkClient->getResponse()->getContent(), true);
+        $this->assertArrayHasKey('jobs', $response, $frameworkClient->getResponse()->getContent());
+        $this->assertEquals([], $response['jobs']);
+        $this->assertEquals(200, $frameworkClient->getResponse()->getStatusCode());
+    }
+
+    public function testProjectStatsDailyMissingFromDate()
+    {
+        $frameworkClient = $this->createClient();
+        $frameworkClient->request(
+            'GET',
+            '/docker/stats/project/daily',
+            [],
+            [],
+            ['HTTP_X-StorageApi-Token' => STORAGE_API_TOKEN]
+        );
+        $response = json_decode($frameworkClient->getResponse()->getContent(), true);
+        $this->assertArrayHasKey('message', $response, $frameworkClient->getResponse()->getContent());
+        $this->assertEquals('Missing "fromDate" query parameter.', $response['message']);
+        $this->assertEquals(400, $frameworkClient->getResponse()->getStatusCode());
+    }
+
+    public function testProjectStatsDailyMissingToDate()
+    {
+        $frameworkClient = $this->createClient();
+        $frameworkClient->request(
+            'GET',
+            '/docker/stats/project/daily?fromDate=2022-08-01',
+            [],
+            [],
+            ['HTTP_X-StorageApi-Token' => STORAGE_API_TOKEN]
+        );
+        $response = json_decode($frameworkClient->getResponse()->getContent(), true);
+        $this->assertArrayHasKey('message', $response, $frameworkClient->getResponse()->getContent());
+        $this->assertEquals('Missing "toDate" query parameter.', $response['message']);
+        $this->assertEquals(400, $frameworkClient->getResponse()->getStatusCode());
+    }
+
+    public function testProjectStatsDailyMissingTimezone()
+    {
+        $frameworkClient = $this->createClient();
+        $frameworkClient->request(
+            'GET',
+            '/docker/stats/project/daily?fromDate=2022-08-01&toDate=2020-01-01',
+            [],
+            [],
+            ['HTTP_X-StorageApi-Token' => STORAGE_API_TOKEN]
+        );
+        $response = json_decode($frameworkClient->getResponse()->getContent(), true);
+        $this->assertArrayHasKey('message', $response, $frameworkClient->getResponse()->getContent());
+        $this->assertEquals('Missing "toDate" query parameter.', $response['message']);
```

Comment:
Tu by malo byt `Missing "timezoneOffset" query parameter.`

Reply:
Hmm, jak to moze prechadzat? Tu vyjimku tam vyhadzujes spravne si myslim.

Reply:
hmm, du se podivat :)

Reply:
hmm, du se podivat :)

Reply:
No on taky neprochazi :)

Reply:
ted uz jo 

![image](https://user-images.githubusercontent.com/4319320/90232276-b9731900-de1c-11ea-94b6-5abb714e5f8c.png)